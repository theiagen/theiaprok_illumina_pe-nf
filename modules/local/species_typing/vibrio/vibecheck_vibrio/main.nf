process VIBECHECK_VIBRIO {
    tag "$meta.id"
    label 'process_single'

    // conda "${moduleDir}/environment.yml"
    // container "docker.io/watronfire/vibecheck:2025.02.24"
    container "wave.seqera.io/wt/b73ba3a93c0e/wave/build:7541bf8cdb9f5a40" //Quick fix for the missing ps tool, see the accompanying Dockerfile

    input:// TODO nf-core: Where applicable all sample-specific information e.g. "id", "single_end", "read_group"
    tuple val(meta), path(reads)
    path(lineage_barcodes)
    output:
    tuple val(meta), path("lineage_report.csv"), emit: report
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def subsampling_fraction = params.vibecheck_subsampling_fraction ? '--subsampling_fraction ${params.vibecheck_subsampling_fraction}' : ''
    def skip_subsampling = params.vibecheck_skip_subsampling ? '--no-detect' : ''
    
    """
    vibecheck ${reads[0]} ${reads[1]} \
        --outdir . \
        ${lineage_barcodes} \
        ${subsampling_fraction} \
        ${skip_subsampling}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        vibecheck: \$(vibecheck --version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    
    touch lineage_report.csv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        vibecheck: \$(vibecheck --version)
    END_VERSIONS
    """
}
