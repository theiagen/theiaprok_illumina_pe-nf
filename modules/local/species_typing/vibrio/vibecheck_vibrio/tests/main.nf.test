// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test vibecheckvibrio
nextflow_process {

    name "Test Process VIBECHECKVIBRIO"
    script "../main.nf"
    process "VIBECHECK_VIBRIO"
    config "./nextflow.config"

    tag "modules"
    tag "vibecheck_vibrio"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("vibrio - reads") {

        // TODO nf-core: If you are created a test for a chained module
        // (the module requires running more than one process to generate the required output)
        // add the 'setup' method here.
        // You can find more information about how to use a 'setup' method in the docs (https://nf-co.re/docs/contributing/modules#steps-for-creating-nf-test-for-chained-modules).

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test_sample' ],
                    [
                        file(params.pipelines_testdata_base_path + "vibrio/data/vib_1.fastq.gz", checkIfExists: true),
                        file(params.pipelines_testdata_base_path + "vibrio/data/vib_2.fastq.gz", checkIfExists: true)
                    ]
                ])
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("vibrio - reads - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test_sample' ],
                    [
                        file(params.pipelines_testdata_base_path + "vibrio/data/vib_1.fastq.gz", checkIfExists: true),
                        file(params.pipelines_testdata_base_path + "vibrio/data/vib_2.fastq.gz", checkIfExists: true)
                    ]
                ])
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }

    }

}
