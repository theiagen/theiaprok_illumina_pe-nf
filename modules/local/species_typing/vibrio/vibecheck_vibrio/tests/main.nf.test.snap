{
    "vibrio - reads - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "lineage_report.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f058d058e7bea244c8f991e71d9bcfad"
                ],
                "report": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "lineage_report.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f058d058e7bea244c8f991e71d9bcfad"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-06-25T14:53:22.980083377"
    },
    "vibrio - reads": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "lineage_report.csv:md5,352077018245847111f9f4fedb85ed3b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f058d058e7bea244c8f991e71d9bcfad"
                ],
                "report": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "lineage_report.csv:md5,352077018245847111f9f4fedb85ed3b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f058d058e7bea244c8f991e71d9bcfad"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-06-25T14:53:12.303511551"
    }
}